
//menerima input pengguna
void main() {
  // var coba = 'percobaan'; //ini adalah variabel didart
  // var number = 21;
  // var varibel_kosong;
  // int coba_number = 12; //number
  // String ad = 'variabel String';
  // double decimal = 1.2;
  // dynamic na = 'tipe apapun';
  // varibel_kosong = 'variabel yang telah diisi';
  // print('Hello World : ${coba} dan ${number}');

  // stdout.write('Nama');
  // String name = stdin.readLineSync()!; //! untuk menandai nilai tidak null
  // stdout.write('usia');
  // int age = int.parse(stdin.readLineSync()!); //konversi input menjadi string
  // print(jsonEncode('nama saya $name usia saya $age tahun'));
    // print(r'Dia baru saja membeli komputer seharga $1,000.00');
    // //String akan memberitahu Dart untuk menganggap String sebagai raw(r)
    // print('Hi (\u2665)');

  // print(5 + 2); // int add = 7
  // print(5 - 2); // int subtract = 3
  // print(5 * 2); // int multiply = 10
  // print(5 / 2); // double divide = 2.5
  // print(5 ~/ 2); //Pembagian, mengembalikan nilai int 2
  // print(5 % 2); // int modulo = 1

// Operasi pembanding
// ==	Sama dengan
// !=	Tidak sama dengan
// >	Lebih dari
// <	Kurang dari
// >=	Lebih dari sama dengan
// <=	Kurang dari sama dengan

//Operator logika
// ||	OR
// &&	AND
// !	NOT

// if (2 <= 3) {
//     print('Ya, 2 kurang dari sama dengan 3');
//   } else {
//     print("Anda salah");
//   }

//     if (2 < 3 && 2 + 4 == 5) {
//     print('Untuk mencetak ini semua kondisi harus benar');
//   } else {
//     print(
//         '2 kurang dari 3, tapi 2 + 4 tidak sama dengan 5, maka ini akan tampil');
//   }

//   if (false || true || false) {
//     print('Ada satu nilai true');
//   } else {
//     print('Jika semuanya false, maka ini akan tampil');
//   }

//Exeption
//untuk mengatasi error exeption yang lebih gunakan try catch dan throw
// try {
//   var a = 7;
//   var b = 0;
//   print(a ~/ b);
// } //on Exception { //adalah pemanfaatan exception secara umum
//   // print('Can not divide by zero.');
//   catch (e,s) {
//     print('Exception happened: $e');
//     print('Stack trace: $s');
//     //tack trace ini kita bisa melihat detail exception dan di baris mana exception tersebut terjadi.
//   // print('Can not divide by zero.');
// } finally {
//   print('This line still executed');
//   //Blok finally akan tetap dijalankan tanpa peduli apa pun hasil yang terjadi pada blok try-catch.
// }


}
